### ISSUE TITLE
Torch.square(a-b)   != a^2 -2ab + b^2

### ISSUE THREAD

**Issue by YqunYang** [pytorch/pytorch#68517]
## To Reproduce

```
import torch

a = torch.rand(1)
b = torch.rand(1)

E1 = a**2-2*a*b+b**2
E2 = torch.square(a-b)

print(E1==E2)
```

## Expected behavior

`> tensor([True])`

## Actual behavior

Big probability is
`> tensor([False])`

Small probability is
`> tensor([True])`

## Environment

This bug is reproduced in three pytorch versions, including 1.7, 1.8, and 1.10.
Other than these three version, I haven't tried the other version.

OS (e.g., Linux): Ubuntu 16.04

## Additional context
It was first found in problem #68422 

Similar bugs may appear in the `torch.abs` `torch.relu` and `torch.sigmoid`.

**Comment by Lezcano** [comment]
This happens because of rounding errors due to finite precision. What it is true is that:
```python
import torch

a = torch.rand(1)
b = torch.rand(1)

E1 = a**2-2*a*b+b**2
E2 = torch.square(a-b)

print(torch.isclose(E1, E2))
```

Closing.

**Comment by YqunYang** [comment]
Thanks
